clifford/_transformation.py

```diff
@@ -0,0 +1,176 @@
+                raise ValueError("MultiVector belongs to the wrong layout")
+            val = val.value
+        else:
+            val = val._matrix
+        self._matrix[to_i, from_i] = val
+
+    def __call__(self, from_item):
+        if from_item.layout != self._layout_from:
+            raise ValueError("Object is from the wrong layout")
+        return self._layout_to.MultiVector(value=self._matrix @ from_item.value)
+
+
+class VectorLinearTransformation(LinearTransformation):
+    r"""
+    A generalization via outermorphisms of a linear transformation to vectors.
+
+    Namely, given a linear transformation :math:`F(u) \to v`, this generalize
+    to the blades by outermorphism, :math:`F(u_1^u_2) \to F(u_1)^F(u_2)`, and to
+    the multivectors by distributivity.
+
+    Such a transformation is grade preserving.
+
+    Arguments
+    ---------
+    matrix : (T, F) array_like
+        A matrix that transforms vectors of size 2**F to size 2**T by
+        left-multiplication.
+    layout_from, layout_to :
+        See :class:`LinearTransformation`
+
+    Example
+    -------
+    We can construct a simple transformation that permutes and non-uniformly
+    scales the basis vectors:
+
+    >>> layout, blades = Cl(3)
+    >>> layout_new, _ = Cl(3, name='f')
+    >>> locals().update(blades)
+    >>> m = np.array([[0, 1, 0],
+    ...               [0, 0, 2],
+    ...               [3, 0, 0]])
+    >>> lt = VectorLinearTransformation(m, layout, layout_new)
+
+    Applying it to some multivectors:
+
+    # the transformation we specified
+    >>> lt(e1), lt(e2), lt(e3)
+    ((3^f3), (1^f1), (2^f2))
+    # the one deduced by outermorphism
+    >>> lt(e12), lt(e23), lt(e13)
+    (-(3^f13), (2^f12), -(6^f23))
+    # and by distributivity
+    >>> lt(1 + e123)
+    1 + (6^f123)
+
+    To inspect the rows and columns:
+
+    # what is the basis blade e_1 in the layout_to
+    >>> lt[:, (1,)]
+    (3^f3)
+    # what is the coefficient of f3 when converting e1 to the layout_to
+    >>> lt[(3,), (1,)]
+    3
+    # no clear meaning, but a consistent interpretation of indexing
+    >>> l[(3,), :]
+    (3e^1)
+    """
+    def __init__(self, matrix, layout_from, layout_to=None):
+        if layout_to is None:
+            layout_to = layout_from
+        if matrix.shape != (layout_to.dims, layout_from.dims):
+            raise ValueError("Matrix should transform vector from layout_from to layout_to")
+        full_matrix = np.zeros((layout_to.gaDims, layout_from.gaDims), dtype=matrix.dtype)
+        super().__init__(full_matrix, layout_from, layout_to)
+        self._vector_matrix = matrix
+
+        # fill the scalar
+        self[(),()] = 1
+
+        # fill the vectors
+        for i_from in range(layout_from.dims):
+            vector_from = (i_from+layout_from.firstIdx,)
+            for i_to in range(layout_to.dims):
+                vector_to = (i_to+layout_to.firstIdx,)
+                self[vector_to, vector_from] = matrix[i_to, i_from]
```

Comment:
Assigning by tuple like this is probably super slow, but optimizing it is a task for later - preferably once we can guarantee the blade order in the `Layout` class.

Reply:
The changes above have been merged since I made this PR - the latest revision makes use of the new features.